src/resolveInlineRef.ts

```diff
@@ -0,0 +1,45 @@
+import { Dictionary } from '@stoplight/types';
+import { extractSourceFromRef } from './extractSourceFromRef';
+import { pointerToPath } from './pointerToPath';
+
+function isObject(maybeObj: unknown): maybeObj is { [key in PropertyKey]: unknown } {
+  return typeof maybeObj === 'object' && maybeObj !== null;
+}
+
+function _resolveInlineRef(document: Dictionary<unknown>, ref: string, seen: Set<unknown>): unknown {
+  const source = extractSourceFromRef(ref);
+  if (source !== null) {
+    throw new ReferenceError('Cannot resolve external references');
+  }
+
+  const path = pointerToPath(ref);
+  let value: unknown = document;
+  for (const segment of path) {
+    if (!isObject(value) || !(segment in value)) {
+      throw new ReferenceError(`Could not resolve '${ref}'`);
```

Comment:
Same thing here - do we want to fail the entire process out for a single bad $ref?

Reply:
See above. I don't think it makes sense for us to preemptively return a value.